libc/include/sys/queue.h

```diff
@@ -35,21 +52,37 @@
 #define SLIST_FOREACH(var, head, field)                                        \
   for ((var) = SLIST_FIRST(head); (var); (var) = SLIST_NEXT(var, field))
 
+#define SLIST_FOREACH_FROM(var, head, field)                                   \
+  for ((var) = ((var) ? (var) : SLIST_FIRST(head)); (var);                     \
+       (var) = SLIST_NEXT(var, field))
```

Comment:
The self assignment is gross. Mind making this:
```c
if (!var) \
  var = SLIST_FIRST(HEAD); \
for (;(var); ... \
```

Reply:
Done.